### PR TITLE
Use primitive types in Position and BlockFace, remove world block con…

### PR DESCRIPTION
**Pull request by creeper123123321**
…nections on Sponge, clean code in 1.8 entity tracker, add constructor for cloning Item and Position

### CODE REVIEW

File: common/src/main/java/us/myles/ViaVersion/protocols/protocol1_13to1_12_2/blockconnections/ConnectionData.java [hunk 1]
```diff
@@ -186,7 +182,11 @@ public static void connectBlocks(UserConnection user, Chunk chunk) {
 
                         ConnectionHandler handler = ConnectionData.getConnectionHandler(block);
                         if (handler != null) {
-                            block = handler.connect(user, new Position(xOff + x, yOff + y, zOff + z), block);
+                            block = handler.connect(user, new Position(
+                                    (int) (xOff + x),
+                                    (short) (yOff + y),
+                                    (short) (zOff + z)
```
Comment (KennyTV) [inline common/src/main/java/us/myles/ViaVersion/protocols/protocol1_13to1_12_2/blockconnections/ConnectionData.java]: shouldn't this be an int?

Comment (creeper123123321) [reply common/src/main/java/us/myles/ViaVersion/protocols/protocol1_13to1_12_2/blockconnections/ConnectionData.java]: Yes

File: common/src/main/java/us/myles/ViaVersion/api/minecraft/Position.java [hunk 2]
```diff
@@ -10,18 +10,59 @@
...
 public class Position {
-    private Long x;
-    private Long y;
-    private Long z;
+    private int posX;
+    private short posY;
+    private int posZ;
+
+    @Deprecated
+    public Position(Long x, Long y, Long z) {
```
Comment (KennyTV) [inline common/src/main/java/us/myles/ViaVersion/api/minecraft/Position.java]: You should go over and replace the usages of this constructor

File: common/src/main/java/us/myles/ViaVersion/api/type/types/minecraft/Position1_14Type.java [hunk 3]
```diff
@@ -17,11 +17,13 @@ public Position read(ByteBuf buffer) {
...
-        return new Position(x, y, z);
+        return new Position((int) x, (short) y, (int) z);
     }
 
     @Override
     public void write(ByteBuf buffer, Position object) {
-        buffer.writeLong(((object.getX() & 0x3ffffff) << 38) | (object.getY() & 0xfff) | ((object.getZ() & 0x3ffffff) << 12));
+        buffer.writeLong((((long) object.getX() & 0x3ffffff) << 38)
+                | (object.getY() & 0xfff)
+                | ((object.getZ() & 0x3ffffff) << 12));
```
Comment (KennyTV) [inline common/src/main/java/us/myles/ViaVersion/api/type/types/minecraft/Position1_14Type.java]: should be
```
        buffer.writeLong((((long) object.getX() & 0x3ffffff) << 38)
                | (object.getY() & 0xfff)
                | (((long) object.getZ() & 0x3ffffff) << 12));
```

File: common/src/main/java/us/myles/ViaVersion/api/type/types/minecraft/PositionType.java [hunk 4]
```diff
@@ -17,11 +17,13 @@ public Position read(ByteBuf buffer) {
...
-        return new Position(x, y, z);
+        return new Position((int) x, (short) y, (int) z);
     }
 
     @Override
     public void write(ByteBuf buffer, Position object) {
-        buffer.writeLong(((object.getX() & 0x3ffffff) << 38) | ((object.getY() & 0xfff) << 26) | (object.getZ() & 0x3ffffff));
+        buffer.writeLong((((long) object.getX() & 0x3ffffff) << 38)
+                | ((object.getY() & 0xfff) << 26)
+                | (object.getZ() & 0x3ffffff));
```
Comment (KennyTV) [inline common/src/main/java/us/myles/ViaVersion/api/type/types/minecraft/PositionType.java]: should be
```
        buffer.writeLong((((long) object.getX() & 0x3ffffff) << 38)
                | ((long) (object.getY() & 0xfff) << 26)
                | (object.getZ() & 0x3ffffff));
```

**Comment by MylesIsCool** [comment]
Ty!

**Pull request merged**